File: core/src/ten_runtime/binding/python/interface/ten_runtime/stop_graph_cmd.py
#
# Copyright © 2025 Agora
# This file is part of TEN Framework, an open source project.
# Licensed under the Apache License, Version 2.0, with certain conditions.
# Refer to the "LICENSE" file in the root directory for more information.
#
from typing import TypeVar, cast

from libten_runtime_python import (
    _StopGraphCmd,  # pyright: ignore[reportPrivateUsage]
    _ten_py_cmd_stop_graph_register_type,  # pyright: ignore[reportPrivateUsage]  # noqa: E501
)

from .cmd import Cmd

T = TypeVar("T", bound="StopGraphCmd")


class StopGraphCmd(_StopGraphCmd, Cmd):
    def __init__(self):  # pyright: ignore[reportMissingSuperCall]
        raise NotImplementedError("Use StopGraphCmd.create instead.")

    @classmethod
    def create(  # pyright: ignore[reportIncompatibleMethodOverride, reportImplicitOverride]  # noqa: E501
        cls: type[T],
    ) -> T:
        return cast(T, cls.__new__(cls))


_ten_py_cmd_stop_graph_register_type(StopGraphCmd)

File: core/src/ten_runtime/binding/nodejs/interface/msg/cmd/stop_graph_cmd.ts
//
// Copyright © 2025 Agora
// This file is part of TEN Framework, an open source project.
// Licensed under the Apache License, Version 2.0, with certain conditions.
// Refer to the "LICENSE" file in the root directory for more information.
//
import ten_addon from "../../ten_addon.js";
import { Cmd } from "./cmd.js";

export class StopGraphCmd extends Cmd {
  private constructor() {
    super("", true);

    ten_addon.ten_nodejs_cmd_stop_graph_create(this);
  }

  static Create(): StopGraphCmd {
    return new StopGraphCmd();
  }

  /**
   * Set the graph ID for this stop graph command.
   */
  setGraphId(graphId: string): void {
    ten_addon.ten_nodejs_cmd_stop_graph_set_graph_id(this, graphId);
  }
}

ten_addon.ten_nodejs_cmd_stop_graph_register_class(StopGraphCmd);
